Caltech-IPAC/firefly
# Firefly-8: Clean up react lifecycle deprecated methods and context

Firefly-8: Clean up react lifecycle deprecated methods and context
   - updated all context to use new context api
   - update many lifecycle method to new style
   - deprecated lifecycle methods that I did not update now start with UNSAFE_
   - payed particular attention to field group

_To test:_

Do _some_ (not all) of the following:
 - try any dialog or panel you are familiar with and make sure it works as expected.
 - Make sure images seem to work
 - Make sure all UI elements works
 - Make sure multi chart viewing works

_To Do:_
 - @tgoldina, @loitly You are welcome to add commits to this pull request:
```bash
  cd firefly/src/firefly/js
  find . -name '*.jsx' -exec fgrep -H 'UNSAFE_' {} \;
```
 - Any file you are familiar with could be updated:
          https://reactjs.org/blog/2018/03/29/react-v-16-3.html
- Some of them should be straight forward but I was not quite sure how to verify if it was working correctly.
